0012-five-bells-ledger-api/0012-five-bells-ledger-api.md

```diff
@@ -0,0 +1,1290 @@
+
+##### Event Types
+[Event Types]: #event-types
+
+A ledger MAY define custom `event` types. A ledger MUST support at least the following `event` values:
+
+| Value             | Resource              | Description                      |
+|:------------------|:----------------------|:---------------------------------|
+| `transfer.create` | [Transfer resource][] | Occurs when a new transfer is prepared. Sent to clients subscribed to the accounts from the `debits` and `credits` fields of the transfer. If the transfer is unconditional, this notification indicates the state of the transaction after execution. The `related_resources` field is omitted. |
+| `transfer.update` | [Transfer resource][] | Occurs when a transfer changes state from `prepared` to `executed` or `rejected`. If the transfer was executed, the `execution_condition_fulfillment` field of `related_resources` MUST contain the fulfillment. If the transfer was rejected, the `related_resources` field is empty. |
+| `message.send`    | [Message resource][]  | Occurs when someone else sends a message. |
+
+If a ledger creates custom event types, their values should follow the convention `{resource}.{event}`.
+
+
+## API Error Codes
+
+The Five Bells Ledger API may return errors using HTTP codes in the range 400-599, depending on the type of error. The message body of the error response is a JSON object with additional information about the nature of the error.
+
+Every error response contains at least the following fields:
+
+| Field     | Type   | Description                                             |
+|:----------|:-------|:--------------------------------------------------------|
+| `id`      | String | A unique error code for this specific type of error, such as `UnmetConditionError`. |
+| `message` | String | A longer, human-readable description for the cause of the error. |
+
+Errors may also have additional arbitrary fields describing the cause or context of the error.
+
+**Note:** Any fields other than `id` and `message` should be considered optional and informational. Clients MUST NOT depend on the presence of those fields. The examples in this spec contain suggestions for additional fields, but those should not be taken as requirements.
+
+### Unauthorized
+[Unauthorized]: #unauthorized
+
+The [authentication information](#authorization-and-authentication) supplied to this request was insufficient for one of the following reasons:
+
+- This method requires authentication but none was provided
+- The credentials were provided using a system not supported by this method (e.g. Basic Auth when Client Certificates are required)
+- The credentials were malformed or don't match the known account information
```

Comment:
> The credentials were malformed or don't match the known account information

Providing wrong credentials for a supported authentication mechanism should result in 403. See: https://en.wikipedia.org/wiki/HTTP_403#Difference_from_status_.22401_Unauthorized.22